core/controllers/base_test.py

```diff
@@ -288,6 +293,25 @@ def test_keys_match_en_qqq(self):
         qqq_key_list = self._extract_keys_from_json_file('qqq.json')
         self.assertEqual(en_key_list, qqq_key_list)
 
+    def test_keys_in_source_code_match_en(self):
+        """Tests that keys in HTML files are present in en.json."""
+        en_key_list = self._extract_keys_from_json_file('en.json')
+        for root, _, files in os.walk(os.path.join(
+                os.getcwd(), self.get_static_asset_filepath())):
```

Comment:
Not sure I understand. Why do you need self.get_static_asset_filepath() -- why not just 'core' and 'extensions'? At the moment, you seem to be checking all of third_party/ as well, which isn't really necessary...